uint8 id
uint16 min_position
uint16 cur_position
uint16 max_position